explorer/client/src/components/search/Search.tsx

```diff
@@ -1,85 +1,183 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import React, { useState, useCallback, useContext } from 'react';
+import React, {
+    useState,
+    useCallback,
+    useContext,
+    useRef,
+    useEffect,
+} from 'react';
 import { useNavigate } from 'react-router-dom';
 
 import { ReactComponent as SearchIcon } from '../../assets/search.svg';
 import { NetworkContext } from '../../context';
-import { navigateWithUnknown } from '../../utils/searchUtil';
+import {
+    navigateWithCategory,
+    SEARCH_CATEGORIES,
+} from '../../utils/searchUtil';
+import { type ResultType } from './SearchResultType';
+import SearchResults from './SearchResults';
 
 import styles from './Search.module.css';
 
 function Search() {
-    const [input, setInput] = useState('');
-    const [network] = useContext(NetworkContext);
     const navigate = useNavigate();
+    const wrapperref = useRef<HTMLDivElement>(null);
+    const [network] = useContext(NetworkContext);
+    const [input, setInput] = useState('');
+
+    const [result, setResult] = useState<ResultType[] | null>(null);
+    const [resultIndex, setResultIndex] = useState(0);
+
+    const handleClickOutside = (event: MouseEvent): void => {
+        if (
+            wrapperref.current &&
+            !wrapperref.current.contains(event.target as Node)
+        ) {
+            setResult(null);
+            setInput('');
+        }
+    };
+
+    const handleKeyPress = useCallback(
+        (event: KeyboardEvent): void => {
+            // Press Down Key or Tab
+            if (event.keyCode === 40 || event.keyCode === 9) {
+                event.preventDefault();
+                setResultIndex((prevIndex) =>
+                    result?.length && prevIndex < result.length - 1
+                        ? prevIndex + 1
+                        : 0
+                );
+            }
+
+            // Press Up Key
+            if (event.keyCode === 38) {
+                event.preventDefault();
+                setResultIndex((prevIndex) =>
+                    prevIndex > 0
+                        ? prevIndex - 1
+                        : result?.length && result.length > 1
+                        ? result.length - 1
+                        : prevIndex
+                );
+            }
+        },
+        [result]
+    );
+
+    // Clicking Outside the Search Bar and Results should clear the search
+
+    useEffect(() => {
+        setResultIndex(0);
+        document.addEventListener('click', handleClickOutside, false);
```

Comment:
Rather than managing this entirely ourself, I think it'd be nice to use something like this headless UI, which can manage some of the hard a11y problems for us.

Ref: https://headlessui.com/react/combobox

Reply:
Thanks for raising the issue of a11y compliance. If there are specific a11y issues that you feel are not being met, this is a question for UX Design and so is best met by a discussion with @mystie711 and perhaps updates to the Figma Design.

Reply:
Regarding more specifically the use of the headlessui library, the problem is that headlessui is opinionated on how key interactions should work. For example, the headlessui library belives the tab key should do the following:

![image](https://user-images.githubusercontent.com/11377188/187417182-93d8c1d6-8b0d-4222-b980-96c64de6e85c.png)

Compare this with the Figma Design:

![image](https://user-images.githubusercontent.com/11377188/187417280-ed563cae-c45f-48a9-8979-f1927a1bfc75.png)

If we use something like headlessui, this would mean major divergences with the Figma Design.